components/gfx/paint_thread.rs

```diff
@@ -392,7 +394,8 @@ impl<C> PaintThread<C> where C: PaintListener + Send + 'static {
             let mut paint_thread = PaintThread {
                 id: id,
                 _url: url,
-                layout_to_paint_port: layout_to_paint_port,
+                layout_to_paint_port:
+                    ROUTER.route_ipc_receiver_to_new_mpsc_receiver(layout_to_paint_port),
```

Comment:
It seems a little arbitrary, and possibly confusing, to do the wrapping here: I think it might be clearer to do it right at the place where `layout_to_paint_port` is originally created -- thus keeping it closer to what the original code did...